main/guides/coreeval/proposal.md

```diff
@@ -0,0 +1,59 @@
+# Code the Proposal
+
+You will need to write a proposal script that runs the contract, and possibly does additional things depending on your needs. (Usually these additional things will be dependent on a governance vote.) For example, [gov-add-psm.js](https://github.com/Agoric/agoric-sdk/blob/master/packages/inter-protocol/test/psm/gov-add-psm.js) is a proposal Agoric created for the PSM contract:
+
+```jsx
+/* global startPSM */
```

Comment:
The `startPSM` global was provided in the pismo boot configuration but isn't provided in vaults.

The global environment for CoreEval scripts is another thing we should document:


https://github.com/Agoric/agoric-sdk/blob/bf164a05f425460ad357841b537b8dbc044357a2/packages/vats/src/core/chain-behaviors.js#L78-L83


where `farExports` comes from `import * as farExports from '@endo/far';` and hence includes `E` and `Far`.
`endowments` comes from a few lines previous:

https://github.com/Agoric/agoric-sdk/blob/bf164a05f425460ad357841b537b8dbc044357a2/packages/vats/src/core/chain-behaviors.js#L40-L45

Documenting `VatData` is a whole other thing; maybe we can leave that out of scope for now.

And `allPowers.modules` comes from:

https://github.com/Agoric/agoric-sdk/blob/bf164a05f425460ad357841b537b8dbc044357a2/packages/vats/src/core/boot-chain.js#L17-L20

`startPSM` was part of `allPowers.modules` in the pismo bootstrap. But in vaults, it's not there.

Reply:
It sounds like you're saying that startPSM isn't a part of CoreEval's global environment anymore. But gov-add-psm.js is checked-in code, it appears to assume that startPSM is part of its global environment, and the code works when I run it. So it seems like startPSM is still in global?

Reply:
> `gov-add-psm.js` is checked-in code

yes, but there are no automated tests for it

> the code works when I run it.

That surprises me. There are no errors in the agd logs? And you can see a new PSM contract started with `agd query vstorage children published.psm.IST`?

Reply:
@michaelfig , @Tyrosine22 says you were there when it worked for him. Any idea how the global reference to `startPSM` worked? I'm confused.

Reply:
Suggestion: just say "The script executes in an environment with globals such as `E` and `Far` provided." and leave enumeration of the whole list of globals to a later PR.

Reply:
> @michaelfig , @Tyrosine22 says you were there when it worked for him.

We verified that the vote passed and that something happened on the chain.

> Any idea how the global reference to `startPSM` worked? I'm confused.

I don't know if the proposal contents were viable or completed successfully, only that they were initiated.

Reply:
The following text was added to the top blurb on proposal.md:

"The script executes in an environment with globals such as E and Far provided."